OpenRA.Mods.Common/Traits/Buildings/Gate.cs

```diff
@@ -0,0 +1,144 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public interface ITemporaryBlocker
+	{
+		bool CanLiftBlockade(Actor self, Actor blocking);
+		bool Blocks();
+	}
+
+	[Desc("Will open and be passable for actors that appear friendly when there are no enemies in range.")]
+	public class GateInfo : BuildingInfo, ITraitInfo, Requires<WithSpriteBodyInfo>
+	{
+		public readonly string OpenSequence = "open";
+		public readonly string ClosingSequence = "closing";
+		public readonly string ClosedSequence = "closed";
+
+		public readonly string OpeningSound = "gateup1.aud";
+		public readonly string ClosingSound = "gatedwn1.aud";
+
+		[Desc("How many ticks until the gate will close after being opened.")]
+		public readonly int TicksToStayOpen = 50;
+
+		[Desc("How many ticks until the gate is open.")]
```

Comment:
will change it, those are the ticks it takes to open the gate.
I made it independent of the actual animation, since ts has different animation lengths for nod/gdi. Gates are considered closed in transitions, that's why it's only used for opening